### DISCUSSION TITLE
Why not use sealed interface over sealed class for children definitions?

### DISCUSSION THREAD

**Discussion by YektaDev** [arkivanov/Decompose#194] [General]
Hi. As you know from the title, I wonder why the examples available in docs use classes in cases [like this](https://arkivanov.github.io/Decompose/child-stack/overview/):
```kotlin
interface Root {

    val childStack: Value<ChildStack<*, Child>>

    sealed class Child {
        class List(val component: ItemList) : Child()
        class Details(val component: ItemDetails) : Child()
    }
}
// ...
```
... when we can instead use:
```kotlin
interface Root {

    val childStack: Value<ChildStack<*, Child>>

    sealed interface Child {
        class List(val component: ItemList) : Child
        class Details(val component: ItemDetails) : Child
    }
}
// ...
```
Is it solely a matter of personal preference or are there other reasons for it?

**Comment by arkivanov** [comment]
I don't remember exactly and can't verify now, but this might be for better interop with Swift. Otherwise, feel free to use sealed interfaces if you like.
